src/ng/compile.js

```diff
@@ -2545,9 +2545,19 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         lastValue,
         parentGet, parentSet, compare;
 
+        if ((attrName in attrs) && !hasOwnProperty.call(attrs, attrName)) {
```

Comment:
Yes. In the case `attrs === {}` and `attrName === 'watch'` we want to make sure code in the switch below accessing attrs['watch'] instead of attrs.prototype.watch. So this is use to find out if `attrName` is in the prototype chain but not actually defined in `attrs` object.